From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sat, 10 Jan 2026 22:03:18 -0300
Subject: [PATCH] Do not load chunks for light checks

Prevents lighting updates from loading unloaded chunks, resolving cascading chunk load issues.

diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 210e2bc69077fd898b126aa00bb5d96f173078ab..70716380551e73516265ce06047717f16b490ecb 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -749,6 +749,8 @@ public abstract class World implements IBlockAccess {
                     blockposition = new BlockPosition(blockposition.getX(), 255, blockposition.getZ());
                 }
 
+                if (!this.isLoaded(blockposition)) return 0; // Paper
+
                 Chunk chunk = this.getChunkAtWorldCoords(blockposition);
 
                 return chunk.a(blockposition, this.I);
